vpdq/cpp/bin/vpdq-hash-video.cpp

```diff
@@ -32,45 +34,39 @@ static void usage(char* argv0, int rc) {
   exit(rc);
 }
 
-// TODO: Move to a shared library
-// Directly copy from TMK
 /**
  *
  * Get the base name with extension of the input path
  *
  * ./dir/sub-dir/sample.txt -> sample.txt
  *
  * @param path Path of the target file
- * @param delimiter The character that marks the beginning or end
  *
  */
-string basename(const string& path, const string& delimiter) {
-  size_t n = path.length();
-  size_t i = path.rfind(delimiter, n);
-  if (i == string::npos) {
+std::string basename(const std::string& path) {
+  size_t i = path.find_last_of("\\/");
+  if (i == std::string::npos) {
     return path;
   } else {
-    return path.substr(i + 1, n - i);
+    return path.substr(i + 1);
   }
 }
 
 /**
  *
- * Stip the extension of the input path
+ * Strip the extension of the input filename
  *
  * .sample.txt -> sample
  *
- * @param path Path of the target file
- * @param delimiter The character that marks the beginning or end
+ * @param filename Path of the target file
  *
  */
-string stripExtension(const string& path, const string& delimiter) {
-  size_t n = path.length();
-  size_t i = path.rfind(delimiter, n);
-  if (i == string::npos) {
-    return path;
+std::string stripExtension(const std::string& filename) {
+  size_t n = filename.rfind('.');
```

Comment:
I removed the delimiter parameter because it's useless. What OS doesn't use "." as the file extension delimiter??

C++17 removes the need for stripExtension() and basename() with `std::filesystem` which provides similar file ops to Python, but this project is C++14.